iocage/cli/activate.py

```diff
@@ -23,20 +23,40 @@
 # POSSIBILITY OF SUCH DAMAGE.
 """activate module for the cli."""
 import click
+import libzfs
 
-import iocage.lib.ioc_common as ioc_common
-import iocage.lib.iocage as ioc
+import iocage.lib.Datasets
+import iocage.lib.Logger
 
 __rootcmd__ = True
 
 
 @click.command(name="activate", help="Set a zpool active for iocage usage.")
 @click.argument("zpool")
-def cli(zpool):
+@click.option("--log-level", "-d", default=None)
+@click.option("--mountpoint", "-m", default="/iocage")
+def cli(zpool, log_level, mountpoint):
     """Calls ZFS set to change the property org.freebsd.ioc:active to yes."""
-    ioc.IOCage(activate=True).activate(zpool)
+    logger = iocage.lib.Logger.Logger(print_level=log_level)
+    zfs = libzfs.ZFS(history=True, history_prefix="<iocage>")
+    iocage_pool = None
 
-    ioc_common.logit({
-        "level"  : "INFO",
-        "message": f"ZFS pool '{zpool}' successfully activated."
-    })
+    for pool in zfs.pools:
+        if pool.name == zpool:
+            iocage_pool = pool
+
+    if iocage_pool is None:
+        logger.error(f"ZFS pool '{zpool}' not found")
+        exit(1)
+
+    try:
+        datasets = iocage.lib.Datasets.Datasets(
+            pool=iocage_pool,
+            zfs=zfs,
+            logger=logger
+        )
+        datasets.activate(mountpoint=mountpoint)
+        logger.log(f"ZFS pool '{zpool}' activated")
+    except:
```

Comment:
Yes, the raise in the next line was a debugging fragment. We want to just exit with 1 if anything goes wrong here. Error output was already sent to the logger from the library at this point.